src/common/commonutils/FileUtils.c

```diff
@@ -58,6 +73,8 @@ static bool SaveToFile(const char* fileName, const char* mode, const char* paylo
 
     if (fileName && mode && payload && (0 < payloadSizeBytes))
     {
+        RestrictFileAccessToCurrentAccountOnly(fileName);

```

Comment:
Is `SaveToFile` only used for existing files, or can the file be created by the `fopen` call that happens later? Looking at it from a logical point of view:
- if the file already exists you're changing the permissions (who says they should be restricted?)
- if the file doesn't exist then the `Restrict` call happens too early and will silently fail

Reply:
Only files that already exist. One core principle for our code is that all files that we write to are restricted for access. The system files that we write to as part of the security baseline have their access permissions set separately.